addons/missileguidance/functions/fnc_shouldFilterRadarHit.sqf

```diff
@@ -0,0 +1,133 @@
+ * 5: Radar seeker angle <NUMBER>
+ *
+ * Return Value:
+ * Whether or not to filter out this target <BOOL>
+ *
+ * Example:
+ * [] call call ace_missileguidance_fnc_shouldFilterRadarHit;
+ *
+ * Public: Yes
+ */
+// as defined by lineIntersectsSurfaces
+#define MAX_LINE_DISTANCE 5000
+
+params ["_target", "_projectile", "_minimumSpeed", "_minimumTime", "_maximumTerrainCheck", "_seekerAngle"];
+TRACE_5("should filter target",_projectile,_target,_minimumSpeed,_minimumTime,_maximumTerrainCheck);
+// Theory developed from doppler radar visuals at https://www.mudspike.com/dcs-f-15c-combat-guide-for-beginners/
+
+// helicopter blades will always produce a doppler shift due to their nature. Don't filter
+if (_target isKindOf "Helicopter" && isEngineOn _target) exitWith {
+    TRACE_2("dont filter helicopters",_target isKindOf "Helicopter",isEngineOn _target);
+    false
+};
+
+private _lineOfSight = (getPosASLVisual _projectile) vectorFromTo (getPosASLVisual _target);
+private _realTargetVelocity = velocity _target;
+private _closingVelocity = _lineOfSight vectorMultiply (_realTargetVelocity vectorDotProduct _lineOfSight);
+private _closingSpeed = vectorMagnitude _closingVelocity;
+
+// if relative target velocity is greather than threshold, we can easily see it. Don't filter
+if (_closingSpeed > _minimumSpeed) exitWith {
+    TRACE_2("dont filter fast objects approaching",_closingSpeed,_minimumSpeed);
+    false
+};
+
+private _projectilePos = getPosASLVisual _projectile;
+
+private _stepDistance = _lineOfSight vectorMultiply (MAX_LINE_DISTANCE min _maximumTerrainCheck);
+
+private _checkPos = _projectilePos;
+private _maskedByGround = false;
+
+// Check for all surfaces until we reach our max range
+for "_i" from 0 to _maximumTerrainCheck step MAX_LINE_DISTANCE do {
+    // determine if target is masked by ground
+    private _endPos = _checkPos vectorAdd _stepDistance;
+    private _groundHit = lineIntersectsSurfaces [_checkPos, _endPos, _projectile, _target];
+
+    _maskedByGround = _groundHit isNotEqualTo [];
+
+    _checkPos = _endPos;
+
+    if (_maskedByGround || (_checkPos select 2) <= 0) then {
+        // for ease assume that we can't check underwater
+        if ((_checkPos select 2) < 0) then {
+            _checkPos set [2, 0];
+        };
+        break
+    };
+};
+
+// looking at sky, target is clear as day. Check for chaff before filtering
+if !(_maskedByGround) exitWith {
+    TRACE_1("dont filter stuff in the sky",_maskedByGround);
+
+    // If there is chaff nearby, check if they will confuse missile
+    private _nearby = _target nearObjects 50;
+    _nearby = _nearby select {
+        // 8 = radar blocking
+        private _blocking = configOf _x >> "weaponLockSystem";
+        private _isChaff = false;
+        if (isNumber _blocking) then {
+            _isChaff = (8 == getNumber _blocking);
+        };
+
+        if (isText _blocking) then {
+            _isChaff = ("8" in getText _blocking);
+        };
```

Comment:
This should account for a number instead of a string (e.g. `24` in config).
```suggestion
        private _isChaff = ([getNumber (configOf _x >> "weaponLockSystem"), 4] call EFUNC(common,binarizeNumber)) select 3;
```